plugins/extractit/icon_extractit/actions/extract_all/action.py

```diff
@@ -4,21 +4,42 @@
 # Custom imports below
 
 from icon_extractit.util.util import Regex
-from icon_extractit.util.extractor import extract, parse_time, clear_emails, extract_filepath, clear_urls, clear_domains
+from icon_extractit.util.extractor import (
+    extract,
+    parse_time,
+    clear_emails,
+    extract_filepath,
+    clear_urls,
+    clear_domains,
+    define_date_time_regex,
+    extract_all_date_formats,
+    parse_time_all_date_formats,
+)
 
 
 class ExtractAll(insightconnect_plugin_runtime.Action):
     def __init__(self):
         super(self.__class__, self).__init__(
-            name="extract_all", description=Component.DESCRIPTION, input=ExtractAllInput(), output=ExtractAllOutput()
+            name="extract_all",
+            description=Component.DESCRIPTION,
+            input=ExtractAllInput(),
+            output=ExtractAllOutput(),
         )
 
     def run(self, params={}):
         string = params.get(Input.STR)
         file = params.get(Input.FILE)
+        date_format = params.get(Input.DATE_FORMAT)
         indicators = {
             "cves": extract(Regex.CVE, string, file),
-            "dates": parse_time(extract(Regex.Date, string, file)),
+            "dates": parse_time(extract(define_date_time_regex(date_format), string, file), date_format)
+            if date_format != "All Formats"
+            else parse_time_all_date_formats(
+                extract_all_date_formats(
+                    string,
+                    file,
+                )
+            ),
```

Comment:
I would possibly split this into its own mini function, or move to a traditional if/else. The ternary here is a bit difficult to read.